components/formats-bsd/test/loci/formats/utests/MemoizerTest.java

```diff
@@ -32,315 +32,201 @@
-      directory.mkdirs();
-      memoizer = new Memoizer(reader, 0, directory);
-
-      // Check existing non-writeable memo directory returns null
-      if (File.separator.equals("/")) {
-        // File.setWritable() does not work properly on Windows
-        directory.setWritable(false);
-        assertEquals(memoizer.getMemoFile(id), null);
-      }
-
-      // Check existing writeable memo diretory returns a memo file
-      directory.setWritable(true);
-      String memoDir = idDir.getAbsolutePath();
-      memoDir = memoDir.substring(memoDir.indexOf(File.separator) + 1);
-      File memoFile = new File(directory, memoDir);
-      memoFile = new File(memoFile, "." + TEST_FILE + ".bfmemo");
-      assertEquals(memoizer.getMemoFile(id).getAbsolutePath(),
-                   memoFile.getAbsolutePath());
+    File directory = createTempDir();
+    Memoizer memoizer = new Memoizer(reader, 0, directory);
+    if (directory.setWritable(false)) {
+      assertNull(memoizer.getMemoFile(id));
+    }
   }
 
   @Test
   public void testGetMemoFilePermissionsInPlaceDirectory() throws Exception {
-      String rootPath = id.substring(0, id.indexOf(File.separator) + 1);
-      memoizer = new Memoizer(reader, 0, new File(rootPath));
-
-      // Check non-writeable file directory returns null for in-place caching
-      if (File.separator.equals("/")) {
-        // File.setWritable() does not work properly on Windows
-        idDir.setWritable(false);
-        assertEquals(memoizer.getMemoFile(id), null);
-      }
-
-      // Check writeable file directory returns memo file beside file
-      idDir.setWritable(true);
-      File memoFile = new File(idDir, "." + TEST_FILE + ".bfmemo");
-      assertEquals(memoizer.getMemoFile(id).getAbsolutePath(),
-        memoFile.getAbsolutePath());
+    Memoizer memoizer = new Memoizer(reader, 0, idDir);
+    if (idDir.setWritable(false)) {
+      assertNull(memoizer.getMemoFile(id));
+    }
   }
 
   @Test
   public void testGetMemoFilePermissionsInPlace() throws Exception {
-      memoizer = new Memoizer(reader);
-
-      // Check non-writeable file directory returns null for in-place caching
-      if (File.separator.equals("/")) {
-        // File.setWritable() does not work properly on Windows
-        idDir.setWritable(false);
-        assertEquals(memoizer.getMemoFile(id), null);
-      }
-      // Check writeable file directory returns memo file beside file
-      idDir.setWritable(true);
-      File memoFile = new File(idDir, "." + TEST_FILE + ".bfmemo");
-      assertEquals(memoizer.getMemoFile(id).getAbsolutePath(),
-        memoFile.getAbsolutePath());
+    Memoizer memoizer = new Memoizer(reader);
+    if (idDir.setWritable(false)) {
+      assertNull(memoizer.getMemoFile(id));
+    }
   }
 
   @Test
   public void testRelocate() throws Exception {
     // Create an in-place memo file
-    memoizer = new Memoizer(reader, 0);
+    Memoizer memoizer = new Memoizer(reader, 0);
     memoizer.setId(id);
```

Comment:
I did not use `checkMemo` because it does the full cycle using the same id, while this test changes id between calls.